### PR TITLE
Firebase custom tokens

### PR DESCRIPTION
**Pull request by mpace965**
Bear with me, because this PR is requires a lot of explanation. Please read before reviewing!

## Storage.go
*  Adds the [firebase admin sdk](https://github.com/firebase/firebase-admin-go), which is used to create Firebase custom tokens for user authentication.
*  Adds [firego](https://github.com/zabawaba99/firego), which is a library for interacting with Firebase via REST in go. It authenticates with our client credentials.
*  `FIREBASE_CONFIG` environment variable. This is a hack, but it is out of necessity. Google's firebase authentication library needs to read the JSON from a file. But, Heroku doesn't have static file storage. So, we must read in the JSON from the environment variable, write it to a file, and then have the Google library read from that file. If you know a better way of doing this, let me know. Ask me on Signal for the value of `FIREBASE_CONFIG`. It is generated from our Firebase client credentials.

## UserManagement.go
*  In the login route, creates a Firebase Custom Token for the user. The [uid](https://firebase.google.com/docs/auth/admin/create-custom-tokens) used is the user's LinkedIn id. Sends the custom token back in the response payload.

## Mobile App
*  Adds [firebase](https://www.npmjs.com/package/firebase) dependency.
*  Adds `fetchFirebaseIdToken` method. For each user in Firebase, we have two tokens. The user's custom token, which they themselves use to sign into firebase, and the user's identity token, which our servers can use to identify the user (using the [VerifyIdToken](https://godoc.org/firebase.google.com/go/auth#Client.VerifyIDToken) method).

Future work needed:
*  Create a Firebase Custom Token refresh endpoint on the backend. Firebase Custom Tokens expire after one hour. Will authenticate with the user's LinkedIn OAuth token.

Also @raokrutarth I would appreciate any Go style comments. I'm seeing now that I should probably switch to tabs in Go files, is that standard?

### CODE REVIEW

File: client/app/screens/LoginScreen.js [hunk 1]
```diff
@@ -71,7 +72,10 @@ class LoginScreen extends React.Component {
 
       const response = await fetch(uri, init);
       const userProfile = await response.json();
-      userProfile = { ...userProfile, isAuthenticated: true }
+      const firebaseIdToken = await fetchIdToken(userProfile.firebaseCustomToken)
+        .catch(err => null);
```
Comment (TomFanella4) [inline client/app/screens/LoginScreen.js:76]: We'll need a better form of error handling here. There are a few instances where errors are "handled" but nothing really happens. I'm thinking of using something like [this](https://docs.nativebase.io/Components.html#Toast) to at least alert the user.

File: client/app/firebase/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  authDomain: "meetoverdb.firebaseapp.com",
+  databaseURL: "https://meetoverdb.firebaseio.com",
+  projectId: "meetoverdb",
+  storageBucket: "meetoverdb.appspot.com",
+  messagingSenderId: FIREBASE_SENDER_ID
+};
+
+const app = firebase.initializeApp(config);
+
+export const fetchIdToken = (token) =>
```
Comment (TomFanella4) [inline client/app/firebase/index.js]: This might be a bit cleaner using the async await syntax.

File: backend/Handlers.go [hunk 3]
```diff
@@ -29,8 +29,9 @@ type ServerResponse struct {
 
 // AuthResponse is the JSON returned to client during login to backend
 type AuthResponse struct {
-	LiProfile   LiProfile      `json:"profile"`
-	AccessToken ATokenResponse `json:"token"`
+	LiProfile            LiProfile       `json:"profile"`
+	AccessToken          ATokenResponse  `json:"token"`
+  FirebaseCustomToken  string          `json:"firebaseCustomToken"`
```
Comment (TomFanella4) [inline backend/Handlers.go]: Tabs vs spaces lol 😄

**Review by TomFanella4 — COMMENTED** [review]
Overall looks good!

File: client/app/screens/LoginScreen.js (hunk 1 above)
Comment (mpace965) [reply client/app/screens/LoginScreen.js:76]: Sounds good. It is at least logged in the implementation of `fetchIdToken`.

File: backend/Storage.go [hunk 4]
```diff
@@ -1,7 +1,68 @@
...
+
+var fbApp     *firebase.App
+var fbClient  *firego.Firebase
+
+func InitializeFirebase() {
+	// This is a Firebase workaround. The Firebase go library MUST read
+	// certificate credentials from a file. Also, Heroku doesn't have static
+	// storage. So, we must create the credential file dynamically from an
+	// environment variable containing the json
+	config := []byte(os.Getenv("FIREBASE_CONFIG"))
```
Comment (raokrutarth) [inline backend/Storage.go:31]: can we not have this file in the backend directory?

**Review by raokrutarth — COMMENTED** [review]
Tabs are the way to go but go fmt should take care of it when I open the files.

File: backend/Storage.go (hunk 4 above)
Comment (TomFanella4) [reply backend/Storage.go:31]: The file contains a private key that must be loaded directly in order to instantiate Firebase. It is a but of a hacky workaround for now. We could possibly create a .env folder or something to store it though.

**Review by AyeeReed — APPROVED** [review]
Looks Great. Thanks for all the work you did to get this to work!

**Pull request merged**